index.js

```diff
@@ -494,13 +537,54 @@ function onStream (client, onerror) {
     }
 
     // All requests to the APM Server must start with a metadata object
+    // The _encodedMetadata property _should_ be set in the Client
+    // constructor function after making a cloud metadata call.
+    //
+    // Since we cork data until the client._encodedMetadata is set the
+    // following conditional should not be necessary. However, we'll
+    // leave it in place out of a healthy sense of caution in case
+    // something unsets _encodedMetadata or _encodedMetadata is somehow
+    // never set.
```

Comment:
That caution could mask a programmer error, no? How about we *assert* if client._encodedMetadata isn't set?

Reply:
Per our discussions I intended to log this maybe-impossible situation as an error -- until I realized the http client does not (appear to?) have access to either its own, or the agent's, logger.  I've opened an issue to tackle this in a separate PR: https://github.com/elastic/apm-nodejs-http-client/issues/132 and will defer for now.